express-notes-api/index.js

```diff
@@ -0,0 +1,100 @@
+const fs = require('fs');
+const dataJson = require('./data.json');
+const express = require('express');
+const app = express();
+
+app.use(express.json());
+
+app.get('/api/notes', (req, res) => {
+  const arrNotes = [];
+  for (const note in dataJson.notes) {
+    arrNotes.push(dataJson.notes[note]);
+  }
+  res.status(200).json(arrNotes);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const idNote = req.params.id;
+  if (idNote < 0) {
+    res.status(400).json({ error: 'id must be a positive integer' });
```

Comment:
In addition to checking whether the id is not negative, please verify it's not zero and that it's an integer.